python_modules/dagster/dagster/core/instance/__init__.py

```diff
@@ -970,6 +987,10 @@ def create_run(
             external_pipeline_origin=external_pipeline_origin,
             pipeline_code_origin=pipeline_code_origin,
         )
+
+        if execution_plan_snapshot:
+            self._log_asset_materialization_planned_events(pipeline_run, execution_plan_snapshot)
+
         return self._run_storage.add_run(pipeline_run)
```

Comment:
might be good to sequence the event writes after the run gets added to the db? incase the run add fails

Reply:
✅